crates/macro/src/derive_props/field.rs

```diff
@@ -83,25 +83,31 @@ impl PropField {
                     #wrapped_name: ::std::option::Option::None,
                 }
             }
-            PropAttr::Default { default } => {
+            PropAttr::PropOr(value) => {
                 let name = &self.name;
-                let ty = &self.ty;
-                let span = default.span();
-                // Hacks to avoid misleading error message.
+                let span = value.span();
                 quote_spanned! {span=>
                     #name: {
                         match true {
-                            #[allow(unreachable_code)]
-                            false => {
-                                let __unreachable: #ty = ::std::unreachable!();
-                                __unreachable
-                            },
-                            true => #default()
+                            false => ::std::unreachable!(),
+                            true => #value,
```

Comment:
Please bring back the hack for here and for `PropOrElse`. It will improve the compiler error messages. It's not great having a confusing message like 

```
"`match` arms have incompatible types"
```

```rust
                let ty = &self.ty;
                let span = value.span();
                // Hacks to avoid misleading error message.
                quote_spanned! {span=>
                    #name: {
                        match true {
                            true => #value,
                            false => {
                                let __unreachable: #ty = ::std::unreachable!();
                            }
                        }
                    },
                }
```

Reply:
Sorry, I made a mistake last time, adding a overly complicated hack. The hack fixed misleading help message like (where `foo` is a function):

```
help: try using a conversion method: `"foo".to_string()`
```

But it introduced another confusing error message:

```
error[E0308]: match arms have incompatible types
```

Here I simplified the hack, but the error message is still there.

Moreover, I just found out that these hacks no longer work, maybe because of some Rust compiler update. Now the error message will contain both "help: try using a conversion method: `foo.to_string()`" and "match arms have incompatible types".

I can't find a way to fix this. So I decide to remove the hack.

For `prop_or`, the help message "try using a conversion method: `123.to_string()`" is not misleading at all, so it does not need the hack in the first place.

Reply:
Sounds good to me!